<a href="https://colab.research.google.com/github/Zekaiiz/MSSP-6070-Zekai-Zhang/blob/main/PARTICIPATION_ACTIVITY_Week_12_Measure_a_p_value_from_a_t_test_on_a_data_set.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
from google.colab import files
uploaded = files.upload()
import pandas as pd
filename = list(uploaded.keys())[0]
df = pd.read_csv(filename, encoding='latin1')
df.head()

Saving Student Performance Data.csv to Student Performance Data (1).csv


,gender,race/ethnicity,parental level of education,lunch,math score,reading score,writing score,total score,test preparation course,admission prospect
0,female,group B,bachelor's degree,standard,72,72,74,218,none,medium
1,female,group C,some college,standard,69,90,88,247,completed,high
2,female,group B,master's degree,standard,90,95,93,278,none,high
3,male,group A,associate's degree,free/reduced,47,57,44,148,none,low
4,male,group C,some college,standard,76,78,75,229,none,high


In [6]:
from scipy import stats
group1 = df[df['gender'] == 'male']['math score']
group2 = df[df['gender'] == 'female']['math score']
t_stat, p_value = stats.ttest_ind(group1, group2, nan_policy='omit')
t_stat, p_value


(np.float64(5.383245869828983), np.float64(9.120185549328822e-08))

In [7]:
from scipy import stats
males = df[df['gender'] == 'male']
prep = males[males['test preparation course'] == 'completed']['math score']
no_prep = males[males['test preparation course'] == 'none']['math score']
t_stat, p_value = stats.ttest_ind(prep, no_prep, nan_policy='omit')

t_stat, p_value

(np.float64(4.222402034670281), np.float64(2.892211700281292e-05))

Q3: “There is a statistically significant difference between the two groups (t = 4.22, p < .001), indicating that the variable being compared has a meaningful relationship with student performance.”